### PR TITLE
Post for tidymodels Q2 update

### PR DESCRIPTION
**Pull request by juliasilge**
This is the second 2️⃣ of our tidymodels roundup posts, summarizing what's happened in Q2.

### CODE REVIEW

**Comment by juliasilge** [comment]
Check out the [preview here](https://deploy-preview-518--tidyverse-org.netlify.app/blog/2021/07/tidymodels-2021-q2/)

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (hfrick) [inline content/blog/tidymodels-2021-q2/index.Rmd:154]:
```suggestion
liblinear_spec <-
```

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (hfrick) [inline content/blog/tidymodels-2021-q2/index.Rmd:162]:
```suggestion
  add_model(liblinear_spec)
```

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (hfrick) [inline content/blog/tidymodels-2021-q2/index.Rmd:72]:
```suggestion
The [butcher](https://butcher.tidymodels.org/) package provides methods to remove (or "axe") components from model objects that are not needed for prediction. There are many methods for various kinds of models, but the most recent release updated how butcher handles _recipes_, the tidymodels' approach for preprocessing and feature engineering, for more complete and robust coverage. Let's consider a simulated churn classification data set for phone company customers:
```

**Comment by hfrick** [comment]
That header pic looks almost like a painting! :paintbrush:

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:41]:
```suggestion
The [tidymodels](https://www.tidymodels.org/) framework is a collection of R packages for modeling and machine learning using tidyverse principles. Earlier this year, we [started regular updates](https://www.tidyverse.org/blog/2021/03/tidymodels-2021-q1/) here on the tidyverse blog summarizing recent developments in the tidymodels ecosystem. You can check out the [`tidymodels tag`](https://www.tidyverse.org/tags/tidymodels/) to find all tidymodels blog posts here, including those that focus on a single package or more major releases. The purpose of these roundup posts is to keep you informed about any releases you may have missed and useful new functionality as we maintain these packages.
```

Comment (batpigandme) [reply content/blog/tidymodels-2021-q2/index.Rmd:41]: "overall" seemed redundant (if you want to keep it, maybe put it before "tidymodels ecosystem"?

I changed the linked text to be more descriptive of the link for accessibility (I am terrible at this, and often make the mistake of linking to the word *here* or something to that effect).

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:43]:
```suggestion
Since our last roundup post, there have been 19 CRAN releases of 15 different packages. That might sound like a lot of change to absorb as a tidymodels user! However, we purposefully write code in small, modular packages that we can release frequently to make models easier to deploy and our software easier to maintain. You can install these updates from CRAN with:
```

Comment (batpigandme) [reply content/blog/tidymodels-2021-q2/index.Rmd:43]: Extraneous comma (aside: I add these little notes when I'm not sure how obvious the change will be in the GitHub UI—let me know if they're simply annoying, and I'll stop)

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:67]: I don't have a hot take on this, but do you think it's worth adding the version number for each item here, since you're linking to the anchored item?

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [reply content/blog/tidymodels-2021-q2/index.Rmd:72]:
```suggestion
The [butcher](https://butcher.tidymodels.org/) package provides methods to remove (or "axe") components from model objects that are not needed for prediction. There are many methods for various kinds of models, but the most recent release updated how butcher handled _recipes_ (the tidymodels approach for preprocessing and feature engineering) for more complete and robust coverage. Let's consider a simulated churn-classification dataset for phone company customers:
```

Comment (batpigandme) [reply content/blog/tidymodels-2021-q2/index.Rmd:72]: I think when you have the "the" in there for "the tidymodels approach" it's not possessive (it would be if it was "tidymodels' approach").

Also, in the following:
> There are many methods for various kinds of models

are you referring to literal R methods? I can't decide if it's one of those special words that we basically can't use in its more quotidian sense because of the context of "being about R." One option would be just to cut that first clause and begin with:
> The most recent release…

I don't think you'd lose much info.

"churn-classification" gets a hyphen just because it's modifying a noun (compound modifier)

Lastly, I think dataset has reached its evolution into closed-form (especially in the [technical context](https://english.stackexchange.com/a/2129/44117)). The two word (open form) -> compound word (hyphenated) -> one word (closed form) evolution is always up for debate (because, y'know, English be like that), but I go closed when it resolves ambiguity _especially_ when the latter term can also be parsed as a verb.

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:90]:
```suggestion
For some kinds of models, we would want to create dummy or indicator variables from nominal predictors, and preprocess features to be on the same scale. We can use recipes for this task:
```

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:108]:
```suggestion
To remove everything from this prepped recipe not needed for applying to new data (e.g. [bake()](https://recipes.tidymodels.org/reference/bake.html) it), we can call `butcher(churn_prep)`. In some applications, modeling practitioners need to make custom functions with a feature-engineering recipe. Sometimes those functions have... a lot of extra STUFF in them, stuff that is not needed for prediction.
```

Comment (batpigandme) [reply content/blog/tidymodels-2021-q2/index.Rmd:108]: compound modifier 🤓

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:129]: I might add a comment above the code here giving the version number of butcher used, or at least say `# old version of butcher` and `# new version of butcher` (or something to that effect). My theory is that some readers pretty much look at the code exclusively, and might not catch the difference (esp since it's shown via `eval = FALSE`/not something they'll reproduce by running the code themselves).

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:148]: Add link to scikit-learn or the relevant C++ lib? (Totally your call—I think it's fine either way).

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:169]: A musing: It's so weird to me that preprocessing has definitively reached closed form, but post-processing is still in compound.

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (batpigandme) [inline content/blog/tidymodels-2021-q2/index.Rmd:187]:
```suggestion
The class predictions created with probably integrate well with functions from yardstick, including custom sets of metrics created with [`metric_set()`](https://yardstick.tidymodels.org/reference/metric_set.html).
```

**Review by batpigandme — APPROVED** [review]
Nicely done, especially with the balance of narrative and examples. My suggestions are minor.

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (DavisVaughan) [inline content/blog/tidymodels-2021-q2/index.Rmd:10]:
```suggestion
    memory footprints for feature-engineering recipes, new model engine options,
```

if we are following @batpigandme's suggestions

**Review by DavisVaughan — APPROVED** [review]

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (juliasilge) [reply content/blog/tidymodels-2021-q2/index.Rmd:67]: I don't think that will add much for the reader? I think I'll pass on that.

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (juliasilge) [reply content/blog/tidymodels-2021-q2/index.Rmd:169]: YES, so weird!

**Comment by juliasilge** [comment]
Thanks to all! 🙏 I'll plan to merge this tomorrow, unless there is any reason not to do so, or to wait until next week.

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (hadley) [inline content/blog/tidymodels-2021-q2/index.Rmd:5]: Include year in title?

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (hadley) [inline content/blog/tidymodels-2021-q2/index.Rmd:62]: I wonder why these aren't getting autolinked in the output

**Review by topepo — APPROVED** [review]

File: content/blog/tidymodels-2021-q2/index.Rmd
Comment (juliasilge) [reply content/blog/tidymodels-2021-q2/index.Rmd]: Great point 👍

**Pull request merged**
